### PR TITLE
Motor profile for ros2_control

### PR DESCRIPTION
**Pull request by livanov93**
# Description
PR introduces support of cia 402 profile using `ros2_control`.

# How to test locally
- [ ] setup virtual can as described [here](https://github.com/ros-industrial/ros2_canopen/blob/e7899ce3cb2fad6071e68e2d96b4b190fd0f1772/canopen/sphinx/quickstart/operation.rst#setup-can-controller)
- [ ] `ros2 launch canopen_ros2_control cia402_system.launch.py`
- [ ] in separate terminal execute: `ros2 service call /cia402_device_1_controller/init std_srvs/srv/Trigger`
- [ ] start position mode by executing: `ros2 service call /cia402_device_1_controller/position_mode std_srvs/srv/Trigger`
- [ ] monitor `/joint_states`  topic to follow the state of the joint: `ros2 topic echo /joint_states`
- [ ] change joint position by executing: 
```
ros2 topic pub /forward_position_controller/commands std_msgs/msg/Float64MultiArray "data:
- 1.0"
```

### CODE REVIEW

**Comment by ipa-cmh** [comment]
@livanov93 
Just ran some tests, works for me. 
Ill try to adapt https://github.com/ipa-cmh/prbt_robot to work with this.

**Comment by ipa-cmh** [comment]
Okay have produced one working version here: https://github.com/ipa-cmh/prbt_robot_moveit_config
Great work.
One question the device controllers are only for a single device right? It would make sense having one that brings up the whole system or at least mutiple joints. Is that possible?

File: canopen_ros2_control/launch/cia402_system.launch.py [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+
+#
+# Author: Lovro Ivanov lovro.ivanov@gmail.com
```
Comment (ipa-cmh) [inline canopen_ros2_control/launch/cia402_system.launch.py:30]: This should be Apache 2.0 no BSD

**Review by ipa-cmh — CHANGES_REQUESTED** [review]
Just one license issue.

**Comment by ipa-cmh** [comment]
@livanov93
Once these changes are made, we can merge this.

**Comment by livanov93** [comment]
> ros2 topic pub /forward_position_controller/commands std_msgs/msg/Float64MultiArray "data:
> - 1.0"

Yes, you are correct here. Currently controllers support one device only. We can create multi-joint controllers which would expose all necessary topics and services per slave node on the bus.

**Comment by livanov93** [comment]
> Okay have produced one working version here: https://github.com/ipa-cmh/prbt_robot_moveit_config
> 
> Great work. One question the device controllers are only for a single device right? It would make sense having one that brings up the whole system or at least mutiple joints. Is that possible?


Yes, you are correct here. Currently controllers support one device only. We can create multi-joint controllers which would expose all necessary topics and services per slave node on the bus.

**Pull request merged**